core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -1070,6 +1075,22 @@ class WorkspaceService(protected val ctx: RawlsRequestContext,
                                                         dataAccess,
                                                         s1
                                 ) { destWorkspaceContext =>
+                                  // TODO: Without limit to make sure all snapshots are fetched
+                                  val snapshotResponse: Future[SnapshotListResponse] = snapshotServiceConstructor(s1)
+                                    .enumerateSnapshots(
+                                      sourceWorkspaceContext.toWorkspaceName,
+                                      0,
+                                      100
+                                    )
+                                  for {
+                                    snapshots <- snapshotResponse
+                                    snapshot <- snapshots.gcpDataRepoSnapshots
+                                  } snapshotServiceConstructor(s1).cloneSnapshotByReference(
+                                    sourceWorkspaceContext.workspaceIdAsUUID,
+                                    destWorkspaceContext.workspaceIdAsUUID,
+                                    snapshot
+                                  )
```

Comment:
I suggest moving all of this code into `SnapshotService`, under a new "clone all snapshot references" method which just accepts the source workspace and destination workspace as arguments. `SnapshotService` should know how to manipulate snapshot references; `WorkspaceService` should be as agnostic as possible about those.